templateflow/api.py

```diff
@@ -63,17 +57,31 @@ def get_metadata(template_id):
 
     """
 
-    filepath = TF_HOME / ('tpl-%s' % template_id) / 'template_description.json'
+    tf_home = Path(TF_LAYOUT.root)
+    filepath = tf_home / ('tpl-%s' % template) / 'template_description.json'
 
     # Ensure that template is installed and file is available
+    if not filepath.is_file():
+        _datalad_get(filepath)
+    return loads(filepath.read_text())
+
+
+def _datalad_get(filepath):
+    if not filepath:
+        return
+
+    try:
+        from datalad import api
+        from datalad.support.exceptions import IncompleteResultsError
+    except ImportError:
+        pass
+
     try:
         api.get(str(filepath))
```

Comment:
This can now raise an `UnboundLocalError` if `from datalad import api` fails. I think letting the `datalad` import fail (or raising an informative exception rather than passing) would be the better move here.